internal/plugin.go

```diff
@@ -622,6 +637,17 @@ func (p *Plugin) applyDefaults(unmarshaledConfig map[string]interface{}) {
 	}
 }
 
+// isDNSCompliant verifies that a string is DNS compliant, as is required by some kubernetes resources
+// https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names
+func isDNSCompliant(name string) bool {
```

Comment:
Do you think you could use this function instead since we already depend on this library?
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/validation#IsDNS1123Subdomain